sos/report/plugins/sunbeam_hypervisor.py

```diff
@@ -0,0 +1,114 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, UbuntuPlugin
+
+
+class SunbeamHypervisor(Plugin, UbuntuPlugin):
+
+    short_desc = "Sunbeam Hypervisor snap"
+
+    plugin_name = "sunbeam_hypervisor"
+    profiles = ('cloud',)
+    packages = ('openstack-hypervisor',)
+
+    common_dir = '/var/snap/openstack-hypervisor/common'
+
+    def setup(self):
+
+        self.add_service_status([
+            'snap.openstack-hypervisor.hypervisor-config-service',
+            'snap.openstack-hypervisor.libvirtd',
+            'snap.openstack-hypervisor.neutron-ovn-metadata-agent',
+            'snap.openstack-hypervisor.nova-api-metadata',
+            'snap.openstack-hypervisor.nova-compute',
+            'snap.openstack-hypervisor.ovn-controller',
+            'snap.openstack-hypervisor.ovs-vswitchd',
+            'snap.openstack-hypervisor.ovsdb-server',
+            'snap.openstack-hypervisor.virtlogd',
+        ])
+
+        self.add_journal('nova-compute')
+
+        self.add_copy_spec([
+            f'{self.common_dir}/*.log',
+            f'{self.common_dir}/log/**/*.log',
+            f'{self.common_dir}/etc',
+            f'{self.common_dir}/lib/nova/instances/*/console.log',
+            f'{self.common_dir}/cache/libvirt/qemu/capabilities/*.xml',
+            f'{self.common_dir}/lib/ovn-metadata-proxy/*.conf',
+        ])
+
+        self.add_forbidden_path([
+            f'{self.common_dir}/etc/ssl/',
+            f'{self.common_dir}/etc/libvirt/secrets',
+            f'{self.common_dir}/etc/libvirt/passwrd.db',
+            f'{self.common_dir}/etc/libvirt/krb5.tab',
+        ])
+
+    def postproc(self):
+
+        # libvirt confs
+        match_exp = r"(\s*passwd=\s*')([^']*)('.*)"
+        libvirt_path_exps = [
+            fr"{self.common_dir}/etc/libvirt/qemu/.*\.xml",
+            fr"{self.common_dir}/etc/libvirt/.*\.conf"
+        ]
+        for path_exp in libvirt_path_exps:
+            self.do_path_regex_sub(path_exp, match_exp, r"\1******\3")
+
+        # nova/neutron bits
+        protect_keys = [
```

Comment:
I think we could collapse this a bit, and get future coverage "for free", by doing something like:

```python
protect_keys = [
    '.*_key',
    '.*_pass(word)?'
]
```

Or otherwise build these keys into the regex.